source/github.js

```diff
@@ -0,0 +1,269 @@
+                        flat.push( repo );
+                        flatIds.push( repo.id );
+                    }
+                } );
+            } );
+            done( null, flat );
+        } );
+    } );
+};
+
+api.getOrgs = function ( accessToken, done )
+{
+    authenticate( accessToken );
+
+    github.user.getOrgs( { per_page: 100 }, done );
+};
+
+var repoPager = function ( url, accessToken, done )
+{
+    var reqOptions = {
+        headers: {
+            "User-Agent": "request"
+        }
+    };
+
+    var totalPages = null;
+
+    var getPage = function ( pageIndex, done )
+    {
+        reqOptions.url = url + "access_token=" + accessToken +  "&per_page=100&page=" + pageIndex;
+
+        request( reqOptions, function ( err, res, body )
+        {
+            if( err || res.statusCode !== 200 )
+            {
+                return done( err );
+            }
+
+            var pageResults = JSON.parse( body );
+
+            if( totalPages )
+            {
+                return done( null, pageResults );
+            }
+
+            totalPages = res.headers.link
+                ? Number( res.headers.link.match( /page=\d*/g ).reverse()[ 0 ].replace( "page=", "" ) )
+                : 1;
+
+            var pageTasks = [];
+            for( var p = 2; p <= totalPages; p++ )
+            {
+                ( function ()
+                {
+                    var pp = p;
+                    pageTasks.push( function ( done )
+                    {
+                        getPage( pp, done );
+                    } );
+                } )();
+            }
+
+            a.parallelLimit( pageTasks, 5, function ( err, results )
+            {
+                var collector = pageResults;
+                results.forEach( function ( r )
+                {
+                    collector = collector.concat( r );
+                } );
+
+                results = results.sort( function ( a, b )
+                {
+                    return a.full_name < b.full_name ? -1 : a.full_name > b.full_name ? 1 : 0;
+                } );
+
+                done( err, collector );
+            } );
+        } );
+    };
+
+    getPage( 1, done );
+};
+
+api.getOwnerRepos = function ( accessToken, done )
+{
+    repoPager( "https://api.github.com/user/repos?", accessToken,  done );
+};
+
+api.getOrgRepos = function ( accessToken, org, done )
+{
+    repoPager( "https://api.github.com/orgs/" + org + "/repos?", accessToken,  done );
+};
+
+api.getHookByUrl = function ( accessToken, owner, repoName, hookUrl, done )
+{
+    authenticate( accessToken );
+
+    var onHooks = function ( err, res )
+    {
+        if( err )
+        {
+            return done ( err );
+        }
+
+        var existingHook = res.filter( function ( r )
+        {
+            return r.config.url === hookUrl;
+        } )[ 0 ];
+
+        if( existingHook )
+        {
+            return done( null, existingHook );
+        }
+
+        return done( null, null );
```

Comment:
This is code that I just moved from another file, but this is a really weird callback, not sure why I wrote it like this when I did, so thanks for noticing.